src/type.cpp

```diff
@@ -627,6 +629,23 @@ const Type *TemplateTypeParmType::GetAsSOAType(int width) const {
     return this;
 }
 
+const Type *TemplateTypeParmType::ResolveDependence(TemplateInstantiation &templInst) const {
+    const Type *resolvedType = templInst.InstantiateType(GetName());
+    if (resolvedType == NULL) {
+        Assert(m->errorCount > 0);
+        return NULL;
+    }
+    Variability v = variability;
+    // TODO: review this.
```

Comment:
Still needs review?

Reply:
It's gone by the final commit in the PR.